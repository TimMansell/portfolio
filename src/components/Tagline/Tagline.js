import React from 'react';

import './Tagline.scss';

export const Tagline = () => {
  return (
    <div className="tagline">
      <div className="tagline__item">Front-end Engineer</div>
      <div className="tagline__item">Javascript Engineer</div>
      <div className="tagline__item">React Engineer</div>
      <div className="tagline__item">VueJS Engineer</div>
    </div>
  );
};

export default Tagline;
