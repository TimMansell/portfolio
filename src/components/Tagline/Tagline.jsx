import styles from './Tagline.module.scss';

export const Tagline = () => {
  return (
    <div className={styles.tagline}>
      <div>Front-end Engineer</div>
      {/* <div className={styles.item}>Javascript Engineer</div>
      <div className={styles.item}>React Engineer</div>
      <div className={styles.item}>VueJS Engineer</div> */}
    </div>
  );
};

export default Tagline;
