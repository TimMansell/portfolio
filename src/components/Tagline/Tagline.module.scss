@import 'variables';
@import 'utilities/pxtorem';
@import 'utilities/text-shadow';

.tagline {
  overflow: hidden;
  opacity: 0;
  animation: tagline 0.5s ease-in-out;
  animation-delay: var(--animation-tagline-delay);
  animation-fill-mode: forwards;
}

.item {
  font-size: pxtorem(32);
  color: var(--secondary-text-colour);
  letter-spacing: pxtorem(1);
  line-height: 1;
  text-transform: uppercase;
  text-shadow: text-shadow(1, 4, var(--colour-1-alt-2));

  @media (min-width: $bp-md) {
    text-shadow: text-shadow(1, 2, var(--tertiary-text-colour));
  }
}

@keyframes tagline {
  from {
    opacity: 0;
    transform: translateY(10px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
}
