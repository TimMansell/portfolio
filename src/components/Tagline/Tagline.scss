@import 'variables';
@import 'utilities/breakpoints';

.tagline {
	height: 30px;
	overflow: hidden;
}