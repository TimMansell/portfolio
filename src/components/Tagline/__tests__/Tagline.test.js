import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Tagline from '../Tagline';

describe("Tagline", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Tagline />);
  });

  it("should match snapshot", () => {
    const snapshot = renderer.create(<Tagline/>).toJSON();
    
    expect(snapshot).toMatchSnapshot();
  });
});
