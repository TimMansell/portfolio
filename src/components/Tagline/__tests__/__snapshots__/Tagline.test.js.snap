// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tagline should match snapshot 1`] = `
<div
  className="tagline"
>
  <h4
    className="tagline__item"
  >
    Front-end Engineer
  </h4>
  <h4
    className="tagline__item"
  >
    Javascript Engineer
  </h4>
  <h4
    className="tagline__item"
  >
    React Engineer
  </h4>
  <h4
    className="tagline__item"
  >
    VueJS Engineer
  </h4>
</div>
`;
